x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/optimizer/EqlFoldSpec.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.eql.optimizer;
+
+import org.elasticsearch.common.Strings;
+
+import java.util.Objects;
+
+public class EqlFoldSpec {
+
+    private final String name;
+    private final String description;
+    private final Boolean caseSensitiveOnly;
```

Comment:
Isn't one `Boolean` enough? That is a test should either be case sensitive, insensitive or null (default).
Otherwise we might have the case where a test is both sensitive and insensitive at the same time.

Reply:
> Otherwise we might have the case where a test is both sensitive and insensitive at the same time.

That's actually intentional. There are some tests written with the correct case in mind (e.g. `process_name  == "System"`) that will work for both configurations. Instead of duplicating those tests, the test is checked against each configuration

Reply:
Isn't that the same as having a `null` value for the `Boolean`? That is, it can be case sensitive, case insensitive or unspecified meaning both.